ansible/roles/common/tasks/os/redhat-8.yml

```diff
@@ -18,11 +18,22 @@
     # name: '*'
     # state: latest
     update_cache: yes
+- name: List all enabled repositories
+  yum:
+    list: repos
+  register: yum_repos
+
+- name: List available repos
+  debug:
+    var: yum_repos.results
 
-- name: Enable CentOS 8 PowerTools Repository
-  command: dnf config-manager --set-enabled powertools
+- name: Enable RHEL 8 CodeReady Repository
+  command: dnf config-manager --set-enabled rhel-8-for-x86_64-appstream-rpms rhel-8-for-x86_64-baseos-rpms codeready-builder-for-rhel-8-x86_64-rpms
+  when: "'rhel-8-for-x86_64-baseos-rpms' in (yum_repos.results | map(attribute='repoid') | map('lower') | list)"
```

Comment:
1. I don't think the two new tasks are necessary. I think the more straightforward way of doing this is `when: ansible_os_distribution == "RedHat"` for the CodeReady repository tasks.
2. You eliminated the PowerTools repository on Rocky 8 systems. Can you please add it back?